src/test/java/com/metamx/common/parsers/RegexParserTest.java

```diff
@@ -0,0 +1,186 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.metamx.common.parsers;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import junit.framework.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ */
+public class RegexParserTest
+{
+  @Test
+  public void testAWSLog()
+  {
+    final String pattern = "^([0-9a-f]+) ([\\w.-]+) \\[([\\w\\/: +-]+)\\] ([\\d.]+) ([^\\s]+) ([\\w]+) ([\\w.-]+) ([^\\s\"]+) \"([^\"]*)\" ([\\d-]+) ([\\w-]+) ([\\d-]+) ([\\d-]+) ([\\d-]+) ([\\d-]+) \"(.+)\" \"(.+)\" ([\\w-]+)$";
+
+    final List<String> fieldNames = Arrays.asList(
+        "Bucket Owner",
+        "Bucket",
+        "Time",
+        "Remote IP",
+        "Requester",
+        "Request ID",
+        "Operation",
+        "Key",
+        "Request-URI",
+        "HTTP status",
+        "Error Code",
+        "Bytes Sent",
+        "Object Size",
+        "Total Time",
+        "Turn-Around Time",
+        "Referrer",
+        "User-Agent",
+        "Version ID"
+    );
+
+    final Parser<String, Object> parser = new RegexParser(
+        pattern,
+        Optional.<String>absent(),
+        fieldNames
+    );
+    String data = "79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be mybucket [06/Feb/2014:00:00:38 +0000] 192.0.2.3 79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be 3E57427F3EXAMPLE REST.GET.VERSIONING - \"GET /mybucket?versioning HTTP/1.1\" 200 - 113 - 7 - \"-\" \"S3Console/0.4\" -";
+
+    final Map<String, Object> parsed = parser.parse(data);
+    ImmutableMap.Builder builder = ImmutableMap.builder();
+    builder.put("Bucket Owner", "79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be");
+    builder.put("Bucket", "mybucket");
+    builder.put("Time", "06/Feb/2014:00:00:38 +0000");
+    builder.put("Remote IP", "192.0.2.3");
+    builder.put("Requester", "79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be");
+    builder.put("Request ID", "3E57427F3EXAMPLE");
+    builder.put("Operation", "REST.GET.VERSIONING");
+    builder.put("Key", "-");
+    builder.put("Request-URI", "GET /mybucket?versioning HTTP/1.1");
+    builder.put("HTTP status", "200");
+    builder.put("Error Code", "-");
+    builder.put("Bytes Sent", "113");
+    builder.put("Object Size", "-");
+    builder.put("Total Time", "7");
+    builder.put("Turn-Around Time", "-");
+    builder.put("Referrer", "-");
+    builder.put("User-Agent", "S3Console/0.4");
+    builder.put("Version ID", "-");
+
+    Assert.assertEquals(
+        "result",
+        builder.build(),
+        parsed
+    );
+  }
+
+  @Test
+  public void testAWSLog2()
```

Comment:
@himanshug it's testing a crazy User-Agent that contains quote marks; agree that a comment would be useful to explain why there are two tests